bitcoin/src/blockdata/script/owned.rs

```diff
@@ -113,14 +113,13 @@ impl ScriptBuf {
                 self.0.push((n % 0x100) as u8);
                 self.0.push((n / 0x100) as u8);
             }
-            n if n < 0x100000000 => {
+            n => {              // `PushBytes` enforces len < 0x100000000
```

Comment:
Would it make sense to have  `debug_assert` to ensure that's the case?

Reply:
To verify buggy code in `PushBytes`?, I don't rekon so.

Reply:
It'd be better to move the responsibility to the type - by having `len_u32` method and having `PushBytes` tested/verified.